mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/ChessFacade.kt

```diff
@@ -199,7 +208,28 @@ private data class StoreMatch(
       }
 
   override suspend fun update(game: Game) {
-    store.collection("games").document(id).update { this["moves"] = game.toAlgebraicNotation() }
+    val chess = store.collection("games").document(id).get<ChessDocument>()
+
+    store.collection("games").document(id).update {
+      this[FieldPath(listOf("metadata", "status"))] =
+          when (game.nextStep) {
+            NextStep.Stalemate -> "stalemate"
+            is NextStep.MovePiece -> null
+            else ->
+                if ((game.nextStep as NextStep.Checkmate).winner == Color.Black) "blackWon"
+                else "whiteWon"
+          }
```

Comment:
What you're looking for is this syntax :

```kotlin
when (val step = game.nextStep) {
  NextStep.Stalemate -> TODO()
  is NextStep.MovePiece -> TODO()
  is NextStep.Checkmate -> // here you can use step.winner, it's smart-casted.
}
```

Reply:
thx, much more cleaner